### DISCUSSION TITLE
View Validation Breaking SQL

### DISCUSSION THREAD

**Discussion by khrome83** [pocketbase/pocketbase#3402] [Q&A] [answered]
Loving Pocketbase and the functionality. But I think I ran into a issue where the view validation (which works really well), breaks more advanced queries. 

I tried to do these two queries (with minor variations, trying to get around validation issues).

### Query 1 - 

```sql
SELECT A.id, 'Stock' AS type, A.label FROM (
  SELECT stock.id as id,(catalog_sets.set_number||"."||characters.name) as label
  FROM stock
  LEFT JOIN inventory ON stock.inventory_id = inventory.id
  LEFT JOIN catalog_sets ON inventory.set_number = catalog_sets.id
  LEFT JOIN characters ON stock.character = characters.id
  WHERE stock.print_label = TRUE
) as A
UNION ALL
SELECT B.id, 'Bundle' AS type, B.label FROM (
  SELECT bundles.id as id, bundles.name as label
  FROM bundles
  WHERE bundles.print_label = TRUE
) as B
```

### Query 2 - 
```sql
WITH stock_labels AS (
  SELECT stock.id as id,(catalog_sets.set_number||"."||characters.name) as label
  FROM stock
  LEFT JOIN inventory ON stock.inventory_id = inventory.id
  LEFT JOIN catalog_sets ON inventory.set_number = catalog_sets.id
  LEFT JOIN characters ON stock.character = characters.id
  WHERE stock.print_label = TRUE
)
, bundle_labels AS (
  SELECT bundles.id as id, bundles.name as label
  FROM bundles
  WHERE bundles.print_label = TRUE
)
SELECT id, 'Stock' AS type, label FROM stock_labels
UNION ALL
SELECT id, 'Bundle' AS type, label FROM bundle_labels
```

Any suggestions? I would love to have this in a view, as opposed to taking it outside of Pocketbase. 

The general idea (ignore complexity on `Stock` table), is just that I create a label name, and filter by what can be printed, and return that, as well as the type, and an ID because views require it. I verified that both queries work for each Stock and Bundle table respectfully. 

Thank you.

**Comment by ganigeorgiev** [comment, marked as answer]
Top-level `UNION` expressions are not supported at the moment and you need to wrap it in a subquery but it comes with some caveats (for more details your could check  https://github.com/pocketbase/pocketbase/discussions/1938).

**Reply by khrome83** [reply]
I really appreciate how fast you have been to respond and your willingness to help folks. You are truly rare in your commitment with an open source project @ganigeorgiev. Thank you for the assist again.

**Comment by khrome83** [comment]
Posting this here so others can see the solution without having to go to #1938 

```sql
SELECT id, type, label 
FROM (
  SELECT stock.id as id, 'Stock' AS type,(catalog_sets.set_number||"."||characters.name) as label
  FROM stock
  LEFT JOIN inventory ON stock.inventory_id = inventory.id
  LEFT JOIN catalog_sets ON inventory.set_number = catalog_sets.id
  LEFT JOIN characters ON stock.character = characters.id
  WHERE stock.print_label = TRUE
  UNION ALL
  SELECT bundles.id as id, 'Bundle' AS type, bundles.name as label
  FROM bundles
  WHERE bundles.print_label = TRUE
)
```
